### PR TITLE
Enable the user to specify a number of concurrent jobs when starting the experiments queue

### PR DESCRIPTION
**Pull request by mattseddon**
# 1/2 `main` <- this <- #3049 

Relates to #3042

This PR puts an input box in front of starting the queue. The user now has to enter the number of workers that they want to run the queue with.

### Demo

https://user-images.githubusercontent.com/37993418/210483080-91e62d71-3d67-4d6e-8fd5-1b1b65b26afe.mov

### CODE REVIEW

File: extension/src/cli/dvc/runner.ts [hunk 1]
```diff
@@ -19,8 +19,7 @@ import { Disposable } from '../../class/dispose'
 
 export const autoRegisteredCommands = {
   EXPERIMENT_RESET_AND_RUN: 'runExperimentReset',
-  EXPERIMENT_RUN: 'runExperiment',
-  EXPERIMENT_RUN_QUEUED: 'runExperimentQueue'
```
Comment (mattseddon) [inline extension/src/cli/dvc/runner.ts:23]: [F] We've move this from here into the executor (no output is returned to the runner's terminal anymore anyway).

Comment (mattseddon) [reply extension/src/cli/dvc/runner.ts:23]: The legacy behaviour was for all of the logs to be fed into the terminal's output.

File: extension/src/test/suite/experiments/workspace.test.ts [hunk 2]
```diff
@@ -439,17 +439,23 @@ suite('Workspace Experiments Test Suite', () => {
...
     it('should be able to execute all experiments in the run queue', async () => {
-      const mockRunExperimentQueue = stub(
-        DvcRunner.prototype,
-        'runExperimentQueue'
-      ).resolves(undefined)
+      const mockQueueStart = stub(DvcExecutor.prototype, 'queueStart').resolves(
+        undefined
+      )
+
+      const dDosNumberOfJobs = '10000'
```
Comment (mattseddon) [inline extension/src/test/suite/experiments/workspace.test.ts:446]: [F] May or may not have run this by accident whilst testing the input box...

File: extension/src/vscode/inputBox.ts [hunk 3]
```diff
@@ -18,3 +18,27 @@ export const getValidInput = (
...
+
+      return ''
+    },
+    options
+  )
+
+  if (!input) {
+    return
+  }
+  return Number.parseInt(input).toString()
```
Comment (mattseddon) [inline extension/src/vscode/inputBox.ts:43]: [F] I'll follow up to bring `setMaxTableHeadDepth` onto this.

**Review by sroy3 — APPROVED** [review]

**Pull request merged**